### ISSUE TITLE
Incorrect error message and error location for invalid escape sequences

### ISSUE THREAD

**Issue by serpent7776** [zaach/jsonlint#142]
Incorrect error message and error location for invalid escape sequences:
```json
{
	"": "foobar\?"
}
```
Current message:
```
Error: Parse error on line 2:
{	"": "foobar\?"}
------^
Expecting 'STRING', 'NUMBER', 'NULL', 'TRUE', 'FALSE', '{', '[', got 'undefined'
```
Expected error message:
```
Invalid escape sequence \?
```
Error location should point directly at `\?`. Current location is especially confusing for very long strings.
